report_qweb_signer/models/report.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import os
+import subprocess
+import tempfile
+import base64
+import time
+from contextlib import closing
+
+from openerp import SUPERUSER_ID, models, api, _
+from openerp.exceptions import Warning, AccessError
+from openerp.tools.safe_eval import safe_eval as eval
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class Report(models.Model):
+    _inherit = 'report'
+
+    def _signer_bin(self, opts):
+        return '/usr/local/bin/PortableSigner -n ' + opts
+
+    def _user_company_id_get(self, cr, uid, context=None):
+        m_users = self.pool.get('res.users')
+        user = m_users.browse(cr, SUPERUSER_ID, uid, context=context)
+        return user.company_id.id
+
+    def _certificate_get(self, cr, uid, ids, report, context=None):
+        if report.report_type != 'qweb-pdf':
+            _logger.info(
+                "Can only sign qweb-pdf reports, this one is '%s' type" %
+                report.report_type)
+            return False
+        m_cert = self.pool.get('report.certificate')
+        company_id = self._user_company_id_get(cr, uid, context=context)
+        certificate_ids = m_cert.search(cr, uid, [
+            ('company_id', '=', company_id),
+            ('model_id', '=', report.model)], context=context)
+        if not certificate_ids:
+            _logger.info(
+                "No PDF certificate found for report '%s'" %
+                report.report_name)
+            return False
+        for cert_id in certificate_ids:
+            cert = m_cert.browse(cr, uid, cert_id, context=context)
+            # Check allow only one document
+            if cert.allow_only_one and len(ids) > 1:
+                _logger.info(
+                    "Certificate '%s' allows only one document, "
+                    "but printing %d documents" %
```

Comment:
This is logging that certificate is configured for signing PDF from one document, but users is printing several documents in one PDF.
My english is not my best skill, feel free to suggest me a more understandable sentence.